### PR TITLE
feat: [PhU-224] add onboarding app

### PR DESCRIPTION
**Pull request by DmytroAlipov**
migr

### YT:
- https://youtrack.raccoongang.com/issue/PhU-224

### Reviewers:
- [ ] @arsentur

### CODE REVIEW

File: requirements/base.txt [hunk 1]
```diff
@@ -6,7 +6,7 @@
 #
 asgiref==3.7.2
     # via django
-django==3.2.21
+# django==3.2.21
```
Comment (arsentur) [inline requirements/base.txt]: Maybe better to remove this?

File: onboarding/views.py
Comment (arsentur) [inline onboarding/views.py:1]: Let's remove this file instead of commenting all inside it.

File: onboarding/urls.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+"""
+The urls for on-boarding app.
+"""
+from django.urls import re_path
+
+from . import views
+
+
+urlpatterns = [
+    # url(r"^onboarding/recommendations/$", views.recommendations, name="recommendations"),
```
Comment (arsentur) [inline onboarding/urls.py]: Let's remove this and leave it as an empty list.

File: onboarding/tests/tests.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        if email_preference:
+            email_preference.delete()
+
+        if user_profile:
+            user_profile.delete()
+
+        if extended_profile:
+            extended_profile.delete()
+
+        NodeBBClient().users.delete_user(username=self.username)
```
Comment (arsentur) [inline onboarding/tests/tests.py]: Seems we removed the usage of NodeBBClient. Do we need to remove it from here?

File: onboarding/serializers.py
Comment (arsentur) [inline onboarding/serializers.py:1]: Let's remove this file.

File: onboarding/settings/common.py [hunk 4]
```diff
@@ -0,0 +1,2 @@
+def plugin_settings(settings):
```
Comment (arsentur) [inline onboarding/settings/common.py]: Common settings file by default has other Django settings. Do we need them here?

File: onboarding/forms.py
Comment (arsentur) [inline onboarding/forms.py:1]: Let's remove this file.

File: onboarding/data/world_languages.json
Comment (arsentur) [inline onboarding/data/world_languages.json:1]: Why do we need this file?

**Review by arsentur — COMMENTED** [review]
In general, all this code needs to be refactored. But we will do it in future MRs or do it as technical debt.

File: requirements/base.txt (hunk 1 above)
Comment (DmytroAlipov) [reply requirements/base.txt]: done

File: onboarding/views.py
Comment (DmytroAlipov) [reply onboarding/views.py:1]: I left the file itself because we need the views.

File: onboarding/tests/tests.py (hunk 3 above)
Comment (DmytroAlipov) [reply onboarding/tests/tests.py]: removed

File: onboarding/serializers.py
Comment (DmytroAlipov) [reply onboarding/serializers.py:1]: done

File: onboarding/forms.py
Comment (DmytroAlipov) [reply onboarding/forms.py:1]: done

File: onboarding/data/world_languages.json
Comment (DmytroAlipov) [reply onboarding/data/world_languages.json:1]: removed it

File: onboarding/admin.py [hunk 5]
```diff
@@ -0,0 +1,160 @@
+from django.contrib import admin
+
+
+# import handlers to make sure that they are executed
+# from . import handlers
```
Comment (dyudyunov) [inline onboarding/admin.py]: @DmytroAlipov please re-check your changes remove such comments

File: onboarding/tests/tests.py [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+        Tests whether all the relevant user data is created
+        """
+        user = User.objects.filter(username=self.username).first()
+        email_preference = EmailPreference.objects.filter(user=user).first()
+        user_profile = UserProfile.objects.filter(user=user).first()
+        extended_profile = UserExtendedProfile.objects.filter(
+            user=user).first()
+
+        data_endpoint = settings.NODEBB_ENDPOINT + '/api/v2/users/data'
+        headers = {'Authorization': 'Bearer ' + settings.NODEBB_MASTER_TOKEN}
```
Comment (dyudyunov) [inline onboarding/tests/tests.py]: We don't have NODEBB and its settings on the project. This should be refactored.

Did you run tests?

**Comment by dyudyunov** [comment]
I agree with Arsen that there is a lot of staff to refactor (later, probably).

I also wonder if we could squash any migrations (maybe all of them without data manipulations e.g. `RunPython` usage). @DmytroAlipov @arsentur what do you think?

There are some useless settings in the oex-plugin app (e.g. sentry initialization), but it is rather out of scope for this task (just a note).

File: onboarding/tests/tests.py (hunk 6 above)
Comment (DmytroAlipov) [reply onboarding/tests/tests.py]: Yes, tests don't work. I just removed these tests for now.

File: onboarding/urls.py (hunk 2 above)
Comment (DmytroAlipov) [reply onboarding/urls.py]: done

File: onboarding/settings/common.py (hunk 4 above)
Comment (DmytroAlipov) [reply onboarding/settings/common.py]: File deleted

File: onboarding/admin.py (hunk 5 above)
Comment (DmytroAlipov) [reply onboarding/admin.py]: done

**Comment by DmytroAlipov** [comment]
@dyudyunov I don't know what to do with the migration squash

**Pull request merged**